[Local Catalog] POS Settings: integrate with catalog data and full sync functionality

jaclync opened the pull request:
Part of WOOMOB-1100
Please review https://github.com/woocommerce/woocommerce-ios/pull/16166 first, or together.

## Description

This PR integrates the settings UI with catalog stats & sync functionality. The implementation includes:

- **POSSettingsLocalCatalogViewModel**: A view model that handles the UI logic for displaying catalog info and triggering full sync from CTA tap
- **Test coverage**: Unit tests for both the view model and service with proper date comparison handling due to different precision in the database

## Steps to reproduce

1. Log in to a store eligible for POS if needed
2. Tap on the POS tab
3. Tap the ellipsis menu > Settings > Catalog --> the catalog size, incremental sync date, full sync date should be accurate
4. In wp-admin, add a product/variation
5. Tap on `Refresh catalog` to start a full sync --> the spinner should be shown in the button. for a site with a small catalog, it should be rather quick. the catalog size & last full sync date should be updated after the sync completes

## Testing information

I tested on a small site, and a large site and left the settings while the full sync was still ongoing.

## Screenshots

Latest copy:

<img width="2360" height="1640" alt="Simulator Screenshot - iPad (A16) - 2025-09-30 at 11 06 42" src="https://github.com/user-attachments/assets/8b957234-454b-4a7c-9188-70b694e72116" />


https://github.com/user-attachments/assets/c10a9211-b451-4063-b917-58578d48ad29




---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Create `POSSettingsLocalCatalogViewModel` that implements catalog settings.

jaclync committed:
Add `POSCatalogSettingsService` that loads data for POS catalog settings.

jaclync committed:
Add test cases for `POSSettingsLocalCatalogViewModel`.

jaclync committed:
Add test cases for `POSCatalogSettingsService`. Date comparison allows a range due to different precision in database.

dangermattic commented:
<!--
  2 Errors: PR requires at least one label...
  2 Warnings: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

jaclync committed:
Fix crashes in `MainTabBarControllerTests` due to fatalError from accessing `ServiceLocator.grdbManager` when the feature flag is disabled.

jaclync commented:
Hi @joshheald no rush reviewing this PR, or just leave initial comments first on a high level review if you get a chance. I will try breaking it into two PRs if you will be reviewing it later.

jaclync committed:
Merge branch 'trunk' into feat/WOOMOB-1100-pos-sync-settings-basic

# Conflicts:
#	WooCommerce/Classes/POS/Utils/PreviewHelpers.swift

jaclync committed:
Combine two catalog sync info getters into one so that only one database call is made.

jaclync committed:
Merge branch 'feat/WOOMOB-1100-pos-sync-settings-yosemite' into feat/WOOMOB-1100-pos-sync-settings-basic

jaclync committed:
Fix lint warning on line length.

jaclync committed:
Show placeholder state when loading catalog info.

jaclync committed:
Merge branch 'feat/WOOMOB-1100-pos-sync-settings-yosemite' into feat/WOOMOB-1100-pos-sync-settings-basic

jaclync committed:
Remove Periphery ignore comment now that the service is integrated.

joshheald reviewed (approved):
Generally looks great, thanks for doing this.

I found that when I switched stores, the old numbers were used. I can't immediately see why, but it only happened if I was quick to open POS, and corrected itself after closing and re-opening POS. So there's some state sticking around between launches. The sync coordinator is suspect here, because it's held on the service locator... but it shouldn't really play a part in getting the numbers, so perhaps there's something else that's not reset when it should be.

Also, we use "Last incremental update" and "Last full sync" as titles, which is inconsistent. I think it would stand out less if we just said "Last update", which is clearer anyway, or "Last incremental sync", though I like that less.

jaclync commented:
> I found that when I switched stores, the old numbers were used. I can't immediately see why, but it only happened if I was quick to open POS, and corrected itself after closing and re-opening POS. So there's some state sticking around between launches. The sync coordinator is suspect here, because it's held on the service locator... but it shouldn't really play a part in getting the numbers, so perhaps there's something else that's not reset when it should be.

I noticed this issue during testing as well, it was from a recent CIAB change p1758613264239289/1758605672.214159-slack-CGPNUU63E. Kiwi had a workaround p1758690950124649/1758605672.214159-slack-CGPNUU63E and I had a PR https://github.com/woocommerce/woocommerce-ios/pull/16173 that refactors this part. I will merge after testing with the latest trunk merged in.

jaclync committed:
Merge branch 'trunk' into feat/WOOMOB-1100-pos-sync-settings-basic

jaclync committed:
Update catalog sync labels from "sync" to "update" based on CR feedback.

jaclync commented:
> Also, we use "Last incremental update" and "Last full sync" as titles, which is inconsistent. I think it would stand out less if we just said "Last update", which is clearer anyway, or "Last incremental sync", though I like that less.

If I understood correctly, I made the following copy changes to show "sync" instead of "sync" in the UI:

- "Last incremental update" → "Last update"
- "Last full sync" → "Last full update" (to match the pattern)

Along with other changes to use "update" for consistency in https://github.com/woocommerce/woocommerce-ios/pull/16144/commits/aba8530f11ca166d52772db7fafa16a6a5a1ea4e. We don't have a design for the settings yet, can make more copy/UI updates later.

<img width="2360" height="1640" alt="Simulator Screenshot - iPad (A16) - 2025-09-30 at 11 06 42" src="https://github.com/user-attachments/assets/34f006bd-2940-4a6a-8989-87667fdfce42" />

jaclync committed:
Fix unit tests from outdated strings.
